packages/medusa-plugin-contentful/src/services/contentful.js

```diff
@@ -594,124 +588,217 @@ class ContentfulService extends BaseService {
-      // check if product exists
-      let variantEntry = undefined
-      // if not, we create a new one
-      try {
-        variantEntry = await environment.getEntry(variant.id)
-      } catch (error) {
-        return this.createProductVariantInContentful(variant)
-      }
+    const v = await this.productVariantService_.retrieve(variant.id, {
+      relations: ["prices", "options"],
+    })
 
-      const v = await this.productVariantService_.retrieve(variant.id, {
-        relations: ["prices", "options"],
-      })
+    const variantEntryFields = {
+      ...variantEntry.fields,
+      [this.getCustomField("title", "variant")]: {
+        "en-US": v.title,
+      },
+      [this.getCustomField("sku", "variant")]: {
+        "en-US": v.sku,
+      },
+      [this.getCustomField("options", "variant")]: {
+        "en-US": this.transformMedusaIds(v.options),
+      },
+      [this.getCustomField("prices", "variant")]: {
+        "en-US": this.transformMedusaIds(v.prices),
+      },
+      [this.getCustomField("medusaId", "variant")]: {
+        "en-US": v.id,
+      },
+    }
 
-      const variantEntryFields = {
-        ...variantEntry.fields,
-        [this.getCustomField("title", "variant")]: {
-          "en-US": v.title,
-        },
-        [this.getCustomField("sku", "variant")]: {
-          "en-US": v.sku,
-        },
-        [this.getCustomField("options", "variant")]: {
-          "en-US": this.transformMedusaIds(v.options),
-        },
-        [this.getCustomField("prices", "variant")]: {
-          "en-US": this.transformMedusaIds(v.prices),
-        },
-        [this.getCustomField("medusaId", "variant")]: {
-          "en-US": v.id,
-        },
+    variantEntry.fields = variantEntryFields
+
+    const updatedEntry = await variantEntry.update()
+    const publishedEntry = await updatedEntry.publish()
+
+    await this.addIgnore_(variant.id, "medusa")
+
+    return publishedEntry
+  }
+
+  async createProductCollectionInContentful(data) {
+    const hasType = await this.getType("collection")
+      .then(() => true)
+      .catch(() => false)
+    if (!hasType) {
+      console.log("couldn't find collection")
+      return Promise.resolve()
+    }
+
+    const collection = await this.productCollectionService_.retrieve(data.id)
+
+    const environment = await this.getContentfulEnvironment_()
+
+    const fields = {
+      [this.getCustomField("medusaId", "collection")]: {
+        "en-US": collection.id,
+      },
+      [this.getCustomField("title", "collection")]: {
+        "en-US": collection.title,
+      },
+      [this.getCustomField("handle", "collection")]: {
+        "en-US": collection.handle,
+      },
+    }
+
+    const result = await environment.createEntryWithId(
+      "collection",
+      collection.id,
+      {
+        fields,
       }
+    )
+
+    return result
+  }
 
-      variantEntry.fields = variantEntryFields
+  async updateProductCollectionInContentful(data) {
```

Comment:
Update Product Collection when a collection is updated in Medusa

Reply:
https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/